libopae/plugins/xfpga/metrics/afu_metrics.c

```diff
@@ -0,0 +1,275 @@
+	return FPGA_NOT_FOUND;
+}
+
+
+fpga_result get_afu_metric_value(fpga_handle handle,
+				fpga_metric_vector *enum_vector,
+				uint64_t metric_num,
+				struct fpga_metric *fpga_metric)
+{
+	fpga_result result                           = FPGA_OK;
+	uint64_t index                               = 0;
+	struct metric_bbb_value metric_csr;
+	struct _fpga_enum_metric *_fpga_enum_metric  = NULL;
+	uint64_t num_enun_metrics                    = 0;
+
+	memset_s(&metric_csr, 0, sizeof(metric_csr));
+
+	if (handle == NULL ||
+		enum_vector == NULL ||
+		fpga_metric == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	result = fpga_vector_total(enum_vector, &num_enun_metrics);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to get metric total");
+		return FPGA_NOT_FOUND;
+	}
+
+
+	for (index = 0; index < num_enun_metrics; index++) {
+
+		_fpga_enum_metric = (struct _fpga_enum_metric *)	fpga_vector_get(enum_vector, index);
+
+		if (metric_num == _fpga_enum_metric->metric_num) {
+
+			result = fpgaReadMMIO64(handle, 0, _fpga_enum_metric->mmio_offset, &metric_csr.csr);
+
+				fpga_metric->value.ivalue = metric_csr.value;
+
+		}
+
+	}
+
+	return result;
+}
+
+fpga_result add_afu_metrics_vector(fpga_metric_vector *vector,
+				  uint64_t *metric_id,
+				  uint64_t group_value,
+				  uint64_t metric_value,
+				  uint64_t metric_offset)
+{
+	fpga_result result                      = FPGA_OK;
+	struct metric_bbb_group group_csr;
+	struct metric_bbb_value metric_csr;
+	char group_name[SYSFS_PATH_MAX]         = { 0 };
+	char metric_name[SYSFS_PATH_MAX]        = { 0 };
+	char qualifier_name[SYSFS_PATH_MAX]     = { 0 };
+	char metric_units[SYSFS_PATH_MAX]       = { 0 };
+
+	memset_s(&group_csr, 0, sizeof(group_csr));
+	memset_s(&metric_csr, 0, sizeof(metric_csr));
+
+	if (metric_id == NULL ||
+		vector == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	group_csr.csr = group_value;
+	metric_csr.csr = metric_value;
+
+	snprintf_s_i(group_name, sizeof(group_name), "%x", group_csr.group_id);
+	snprintf_s_i(metric_name, sizeof(metric_name), "%x", metric_csr.counter_id);
+
+	sprintf(qualifier_name, "%s:%x:%x", "AFU", group_csr.group_id, metric_csr.counter_id);
+	snprintf_s_i(metric_units, sizeof(metric_units), "%x", group_csr.units);
+
+	*metric_id = *metric_id + 1;
+
+	add_metric_vector(vector, *metric_id, qualifier_name, group_name, "",
+		metric_name, "", metric_units, FPGA_METRIC_DATATYPE_INT, FPGA_METRIC_TYPE_AFU, FPGA_HW_MCP, metric_offset);
+
+	return result;
+}
+
+
+fpga_result enum_afu_metrics(fpga_handle handle,
+			    fpga_metric_vector *vector,
+			    uint64_t *metric_id,
+			    uint64_t metrics_offset)
+{
+	fpga_result result                    = FPGA_NOT_FOUND;
+	struct metric_bbb_group group_csr;
+	struct metric_bbb_value metric_csr;
+	uint64_t value_offset                 = 0;
+	uint64_t group_offset                 = 0;
+
+	memset_s(&group_csr, 0, sizeof(group_csr));
```

Comment:
The signature of memset_s is..
```c
errno_t memset_s(void *dest, rsize_t len, uint8_t value);
```
This call isn't correct.  Please fix.

Reply:
fixed